core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -172,6 +247,8 @@ private[spark] class ChunkedByteBuffer(var chunks: Array[ByteBuffer]) {
 
 private[spark] object ChunkedByteBuffer {
 
+  val COPY_BUFFER_LEN: Int = 1024 * 1024
```

Comment:
Might be messy to add a dependency from here to `SerializerHelper` - add a comment in both places to indicate they should be ideally kept in sync ?

Reply:
I added a `def estimateBufferChunkSize(estimatedSize: Long = -1)` to be used for both. But I'm not sure if the heuristic is appropriate.....

Or another option: we can use `1024`(1KB) for all, make it simple. I did some quick benchmark, 1KB isn't too bad compared to 1MB even in large result, and the overhead upper bound is reasonable even when result is very tiny(actually even a nearly empty result will still be serialized to a few hundred bytes because of other metrics and accumulators)

WDYT?

Reply:
Added a comment which might help ...